app/controllers/concerns/castle_tracking.rb

```diff
@@ -0,0 +1,12 @@
+module CastleTracking
+  extend ActiveSupport::Concern
+
+  included do
+    def track_castle_event(castle_event, user)
+      context = ::Castle::Client.to_context(request)
```

Comment:
IMHO, it would much nicer if we could pass just what was needed instead of the entire request object. As far as I can tell, only following items are needed:
```md
## headers
- 'X-Castle-Client-Id'
- Accept-Language
- User-Agent
- X-Forwarded-For Client-Ip Remote-Add

## cookie 
- '__cid'
```
Please mention any other item I may have missed. Passing the entire object means every time we update this gem we will have to go through the entire changeset very closely. 

Let's add `Castle.config.whitelisted`. Also, is it possible to not send client_ip as is and instead hash it?